src/test/java/ch/wisv/areafiftylan/OrderRestIntegrationTest.java

```diff
@@ -924,6 +924,43 @@ public void testGetTicketAvailability(){
         then().
             body("ticketType", hasItems(equalTo("EARLY_FULL"), equalTo("REGULAR_FULL")));
     }
+
+    @Test
+    public void testGetOneValidTicket() {
+        insertTestOrders();
+        SessionData login = login("user");
+
+        Ticket ticket = new Ticket(user, TicketType.EARLY_FULL, false, false);
+        ticket.setValid(true);
+        ticketRepository.saveAndFlush(ticket);
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            get("/users/current/tickets").
+        then().
+            body("object", hasSize(1)).
+            body("[0].owner.username", is("user"));
+        //@formatter:on
+    }
+
+    @Test
+    public void testZeroValidTickets() {
+        insertTestOrders();
+        SessionData login = login("user");
+
+        //@formatter:off
+        given().
+            filter(sessionFilter).
+            header(login.getCsrfHeader()).
+        when().
+            get("/users/current/tickets").
+        then().
+            body("object", hasSize(0));
+        //@formatter:on
+    }
 }
```

Comment:
You're partially right. See the GET method for the way to do this and why

```
// To prevent 403 errors on this endpoint, we manually handle unauthenticated users, instead of a
        // preauthorize tag.
        if (auth != null) {
            // Get the currently logged in user from the autowired Authentication object.
            UserDetails currentUser = (UserDetails) auth.getPrincipal();
            User user = userService.getUserByUsername(currentUser.getUsername()).get();
```